source/unified-test-format/unified-test-format.rst

```diff
@@ -2492,7 +2560,224 @@ An example of this operation follows::
+to be run on a given thread. runOnThread MUST NOT wait for the scheduled
+operation to complete. If any of the operation's test assertions fail, the
+entire test case MUST fail as well.
+
+The following arguments are supported:
+
+- ``thread``: Required string. Thread entity on which this operation should be
+  scheduled.
+
+- ``operation``: Required `operation`_ object. The operation to schedule on the
+  thread. This object must be a valid operation as described in `Entity Test
+  Operations`_.
+
+An example of this operation follows::
+
+     - name: runOnThread
+       object: testRunner
+       arguments:
+         thread: *thread0
+         operation:
+           name: insertOne
+           object: *collection0
+           arguments:
+             document: { _id: 2 }
+           expectResult:
+             $$unsetOrMatches:
+               insertedId: { $$unsetOrMatches: 2 }
+
+
+waitForThread
+~~~~~~~~~~~~~
+
+The "waitForThread" operation instructs the test runner to notify the given
+thread that no more operations are forthcoming, wait for it to complete its last
+operation, and assert that it exited without any errors.
+
+The following arguments are supported:
+
+- ``thread``: Required string. Thread entity that should be stopped and awaited
+  for completion.
+
+An example of this operation follows::
+
+    - name: waitForThread
+      object: testRunner
+      arguments:
+        thread: *thread0
+  
+
+waitForEvent
+~~~~~~~~~~~~
+
+The "waitForEvent" operation instructs the test runner to wait until the
+specified MongoClient has published a specific, matching event a given number of
+times. Note that this includes any events published before the waitForEvent
+operation started.
+
+The following arguments are suported:
+
+- ``client``: Required string. Client entity whose events the runner will be
+  waiting for.
+
+- ``event``: Required `expectedEvent`_ object. The event which the test runner
+  is waiting to be observed on the provided client. The assertions laid out in
+  `expectedEvent`_ MUST be used to determine if an observed event matches
+  ``event``. ``event`` SHOULD have an event type that was included in the
+  ``client``'s ``observeEvents`` field.
+
+- ``count``: Required integer. The number of matching events to wait for before
+  resuming execution of subsequent operations.
+
+For example, the following instructs the test runner to wait for at least one
+poolClearedEvent to be published::
+
+    - name: waitForEvent
+      object: testRunner
+      arguments:
+        client: *client0
+        event:
+          poolClearedEvent: {}
+        count: 1
+
+
+assertEventCount
+~~~~~~~~~~~~~~~~
+
+The "assertEventCount" operation instructs the test runner to assert the
+specified MongoClient has published a specific, matching event a given number of
+times so far in the test. Note that this includes any events published before
+the waitForEvent operation started.
```

Comment:
The reference to "waitForEvent" may be copypasta. If `assertEventCount` is intended to be an instantaneous check, I don't think this sentence is needed.

Alternatively, would it make sense to consolidate `assertEventCount` and `waitForEvent` into one operation? For example, if you add a `waitMS` option that defaults to zero (i.e. instantaneous check of previously recorded events), then you could specify a positive number when you'd like to wait for forthcoming events.

The downside as I see it is if `waitForEvent` is widely used, you'd be repeating the timeout period throughout the tests and there wouldn't be an opportunity to have a configurable timeout _and_ default to a non-zero period (like the ten second timeout you have a for `waitForPrimaryChange`).

Reply:
I could see a consolidation working, but I think for ease-of-conversion from the original tests which had both operations, I'd prefer to keep them as separate operations.

And yeah that was copypasta, removed.